README.md

```diff
@@ -1,636 +1,539 @@
+    │   └── vpc
+    │       └── main.tf
+    ├── qa
+    │   ├── app
+    │   │   └── main.tf
+    │   ├── mysql
+    │   │   └── main.tf
+    │   └── vpc
+    │       └── main.tf
+    └── stage
+        ├── app
+        │   └── main.tf
+        ├── mysql
+        │   └── main.tf
+        └── vpc
+            └── main.tf
 ```
 
-Terragrunt forwards almost all commands, arguments, and options directly to Terraform, using whatever version of
-Terraform you already have installed. However, before running Terraform, Terragrunt will ensure your remote state is
-configured according to the settings in the `terraform.tfvars` file. Moreover, for the `apply`, `refresh`, and 
-`destroy` commands, Terragrunt will first try to acquire a lock using [DynamoDB](#locking-using-dynamodb):
+The contents of each environment will be more or less identical, except perhaps for a few settings (e.g. the prod
+environment may run bigger or more servers). As the size of the infrastructure grows, having to maintain all of this  
+duplicated code between environments becomes more error prone. You can reduce the amount of copy paste using 
+[Terraform modules](https://blog.gruntwork.io/how-to-create-reusable-infrastructure-with-terraform-modules-25526d65f73d), 
+but even the code to instantiate a module and set up input variables, output variables, providers, and remote state 
+can still create a lot of maintenance overhead.  
 
-```
-terragrunt apply
-[terragrunt] 2016/05/30 16:55:28 Configuring remote state for the s3 backend
-[terragrunt] 2016/05/30 16:55:28 Running command: terraform remote config -backend s3 -backend-config=key=terraform.tfstate -backend-config=region=us-east-1 -backend-config=encrypt=true -backend-config=bucket=my-bucket
-Initialized blank state with remote state enabled!
-[terragrunt] 2016/05/30 16:55:29 Attempting to acquire lock for state file my-app in DynamoDB
-[terragrunt] 2016/05/30 16:55:30 Attempting to create lock item for state file my-app in DynamoDB table terragrunt_locks
-[terragrunt] 2016/05/30 16:55:30 Lock acquired!
-[terragrunt] 2016/05/30 16:55:30 Running command: terraform apply
-terraform apply
+How can you keep your Terraform code [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) so that you only 
+have to define it once, no matter how many environments you have?
 
-aws_instance.example: Creating...
-  ami:                      "" => "ami-0d729a60"
-  instance_type:            "" => "t2.micro"
 
-[...]
+#### Remote Terraform configurations
 
-Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
+Terragrunt has the ability to download remote Terraform configurations. The idea is that you define the Terraform code 
+for all of your infrastructure in a single repo, called, for example, `modules`:
 
-[terragrunt] 2016/05/27 00:39:19 Attempting to release lock for state file my-app in DynamoDB
-[terragrunt] 2016/05/27 00:39:19 Lock released!
+```
+└── modules
+    ├── prod
+    │   ├── app
+    │   │   └── main.tf
```

Comment:
Doesn't this pattern still end up with a lot of copy and paste between the resources defined in `prod/app/main.tf` and `otherenv/app/main.tf`?  Does the `modules` repo actually need to be environment specific?